### PR TITLE
Docs: Icon specs for Android

### PR DESCRIPTION
**Pull request by hectoid**
Published a draft of Android icon specs. Made a minor language addition to web Link do's and dont's.

Thanks for creating a PR! Please follow this template and delete items/sections that are not relevant to your changes, including these instructions.

Please also make sure your [PR title](https://github.com/pinterest/gestalt/#releasing) matches our format: `{ComponentName}-{platform}: Description`

### Summary
Mainly adding Figma specs to Android Icon component page.

#### What changed?

Added specs for icons via Figma embeds. Re-arranged order to accommodate the specs

#### Why?

Android developers asked that we include figma specs in the documentation

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)](https://jira.pinadmin.com/browse/GESTALT-5679)

### Checklist

- yes Added unit and Flow Tests
- yes Added documentation + accessibility tests
- na Verified accessibility: keyboard & screen reader interaction
- yes Checked dark mode, responsiveness, and right-to-left support
- waiting for feedback Checked stakeholder feedback (e.g. Gestalt designers)
image

### CODE REVIEW

File: docs/markdown/android/icon.md [hunk 1]
```diff
@@ -94,6 +86,48 @@ Used sparingly to draw attention to an icon that might otherwise be missed.
 </Group>
 </TwoCol>
 
+### Color
+Icon colors are semantic—they have a specific meaning and aren't arbitrary. There is no disabled color for icons, as that is handled by the button state that an icon is in.
```
Comment (AlbertCarreras) [inline docs/markdown/android/icon.md:90]: semantic —they

typo

File: docs/markdown/android/icon.md [hunk 2]
```diff
@@ -94,6 +86,48 @@ Used sparingly to draw attention to an icon that might otherwise be missed.
...
+</Group>
+<Group>
+<ImgContainer src="https://i.pinimg.com/originals/e8/a4/1c/e8a41cc5c6ce014dc50bfc801ca7fb01.jpg" alt="A arrow pointing left." />
+directional-arrow-left
+</Group>
+</ThreeCol>
+
+## Naming
+
+* Follow the web icon naming convention when possible
```
Comment (AlbertCarreras) [inline docs/markdown/android/icon.md:126]: icon naming convention< can we add a link to this conventions?

File: docs/markdown/android/icon.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ fullwidth: true
 <ImgContainer src="https://i.pinimg.com/originals/5c/e9/65/5ce965aac16951ef57b3a735bb41e823.jpg" alt="Four example icons in a row. Icons include share, check, back, and menu." />
 
 
+## Icon assets
+
+Download svg assets from our [Android icon GitHub repo](packages/gestalt/src/icons-android).
```
Comment (AlbertCarreras) [inline docs/markdown/android/icon.md:11]: svg >> SVG typo

This link goes nowhere.

**Comment by AlbertCarreras** [comment]
The library is still no available, right?

File: docs/markdown/android/icon.md (hunk 3 above)
Comment (AlbertCarreras) [reply docs/markdown/android/icon.md:11]: Android icon GitHub repo what is this?

File: docs/markdown/android/icon.md (hunk 1 above)
Comment (hectoid) [reply docs/markdown/android/icon.md:90]: thank you! what's the typo? a space before the em dash? Per content standards, no spaces: 
image
https://gestalt.pinterest.systems/foundations/content_standards/grammar#Punctuation

File: docs/markdown/android/icon.md (hunk 2 above)
Comment (hectoid) [reply docs/markdown/android/icon.md]: good idea! added! since this is markdown, I can't currently include our "internal only icon".

**Review by dangerismycat — COMMENTED** [review]
Ok, I did lead you astray. We don't want to include these icons in `packages/gestalt/src`, as then they'll be needlessly bundled with the rest of the component library and bloat bundle sizes. Though we're not exporting them as a separate package yet, let's move these to `packages/gestalt-icons-android`. (You should be able to accomplish this in one go by renaming the directory in VS Code — this should be super easy, not a tedious slog.) My apologies for the thrash! Otherwise this PR looks great 🙂

File: docs/markdown/android/icon.md (hunk 3 above)
Comment (hectoid) [reply docs/markdown/android/icon.md]: a temporary folder for icons until they can automate the process

File: docs/markdown/android/icon.md [hunk 4]
```diff
@@ -40,38 +44,26 @@ People use Android's accessibility features, such as TalkBack and dynamic text s
 [Accessible design on Android](https://material.io/design/usability/accessibility.html#understanding-accessibility)
 [Accessible development on Android](https://developer.android.com/guide/topics/ui/accessibility)
 
-## Variants
+## Design tokens
+Use these tokens for applying size and color styles to icon.
 
-### Styling
-For information on colors and states, refer to the [web Icon guidelines](/web/icon).
+<iframe style={{border:0}} width="100%" height="692" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FREw1COFYAktmVWrUBh3Ov8%2FGestalt-for-Android%3Fnode-id%3D5057%253A18721%26t%3D5FK5Q8asTy0922qM-1" allowfullscreen></iframe>
```
[the remark this answers is not in the document]
Comment (hectoid) [reply docs/markdown/android/icon.md:50]: they look fine enough given the figma embed limitations. we are waiting to get approval to publish these publicly. 
image

I fixed the folder path.

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**